app/validators/answer_code_validator.py

```diff
@@ -54,17 +54,17 @@ def validate_duplicates(self):
 
         if len(duplicates) > 0:
```

Comment:
To be consistent with the one below, this could just be `if duplicates`. I would update the var names to be obvious.

I suggest simplifying to something like this, since the check is the same for both.

```py
    def validate_duplicates(self):
        answer_ids = [
            answer_code["answer_id"]
            for answer_code in self.answer_codes
            if "answer_value" not in answer_code
        ]

        for values, error_message in [
            (self.codes, self.DUPLICATE_ANSWER_CODE_FOUND),
            (answer_ids, self.DUPLICATE_ANSWER_ID_FOUND),
        ]:

            duplicates = find_duplicates(values)

            if duplicates:
                self.add_error(error_message, duplicates=list(duplicates))
```

Reply:
Updated 👍